vector/src/main/java/im/vector/app/features/command/CommandParser.kt

```diff
@@ -361,57 +353,49 @@ object CommandParser {
-                                split.subList(1, split.size)
+                                messageParts[1],
+                                messageParts.drop(2)
                         )
+                    } else {
+                        ParsedCommand.ErrorSyntax(Command.CREATE_SPACE)
                     }
                 }
-                Command.ADD_TO_SPACE.command                 -> {
-                    val rawCommand = textMessage.substring(Command.ADD_TO_SPACE.command.length).trim()
-                    ParsedCommand.AddToSpace(
-                            rawCommand
-                    )
-                }
-                Command.JOIN_SPACE.command                   -> {
-                    val spaceIdOrAlias = textMessage.substring(Command.JOIN_SPACE.command.length).trim()
-                    ParsedCommand.JoinSpace(
-                            spaceIdOrAlias
-                    )
-                }
-                Command.LEAVE_ROOM.command                   -> {
-                    val spaceIdOrAlias = textMessage.substring(Command.LEAVE_ROOM.command.length).trim()
-                    ParsedCommand.LeaveRoom(
-                            spaceIdOrAlias
-                    )
-                }
-                Command.UPGRADE_ROOM.command                 -> {
-                    val newVersion = textMessage.substring(Command.UPGRADE_ROOM.command.length).trim()
-                    if (newVersion.isEmpty()) {
-                        ParsedCommand.ErrorSyntax(Command.UPGRADE_ROOM)
+                Command.ADD_TO_SPACE.matches(slashCommand)                 -> {
+                    ParsedCommand.AddToSpace(spaceId = message)
+                }
+                Command.JOIN_SPACE.matches(slashCommand)                   -> {
+                    ParsedCommand.JoinSpace(spaceIdOrAlias = message)
+                }
+                Command.LEAVE_ROOM.matches(slashCommand)                   -> {
+                    ParsedCommand.LeaveRoom(roomId = message)
+                }
+                Command.UPGRADE_ROOM.matches(slashCommand)                 -> {
+                    if (message.isNotEmpty()) {
+                        ParsedCommand.UpgradeRoom(newVersion = message)
                     } else {
-                        ParsedCommand.UpgradeRoom(newVersion)
+                        ParsedCommand.ErrorSyntax(Command.UPGRADE_ROOM)
                     }
                 }
-                else                                         -> {
+                else                                                       -> {
                     // Unknown command
                     ParsedCommand.ErrorUnknownSlashCommand(slashCommand)
                 }
             }
         }
     }
+
+    private fun trimParts(message: CharSequence, messageParts: List<String>): String? {
+        val partsSize = messageParts.sumOf { it.length }
+        val gapsNumber = messageParts.size - 1
```

Comment:
I am wondering if it still works well if the user adds more space chars between the params